contracts/external/oracles/ChainlinkAutomationPriceOracle.sol

```diff
@@ -0,0 +1,179 @@
+
+    constructor(
+        address _dolomiteMargin,
+        address _chainlinkRegistry
+    ) OnlyDolomiteMargin(_dolomiteMargin) {
+        CHAINLINK_REGISTRY = _chainlinkRegistry;
+    }
+
+    function ownerSetHeartbeat(uint256 _heartbeat) external onlyDolomiteMarginOwner(msg.sender) {
+        _ownerSetHeartbeat(_heartbeat);
+    }
+
+    function ownerSetGracePeriod(uint256 _gracePeriod) external onlyDolomiteMarginOwner(msg.sender) {
+        _ownerSetGracePeriod(_gracePeriod);
+    }
+
+    function ownerSetUpperEdge(uint256 _upperEdge) external onlyDolomiteMarginOwner(msg.sender) {
+        _ownerSetUpperEdge(_upperEdge);
+    }
+
+    function ownerSetLowerEdge(uint256 _lowerEdge) external onlyDolomiteMarginOwner(msg.sender) {
+        _ownerSetLowerEdge(_lowerEdge);
+    }
+
+    function ownerSetChainlinkRegistry(address _chainlinkRegistry) external onlyDolomiteMarginOwner(msg.sender) {
+        _ownerSetChainlinkRegistry(_chainlinkRegistry);
+    }
+
+    function checkUpkeep(bytes calldata /* checkData */)
+    external
+    returns (bool upkeepNeeded, bytes memory /* performData */) {
+        Require.that(
+            tx.origin == address(0),
+            _FILE,
+            "static rpc calls only"
+        );
+
+        return (_checkUpkeepConditions(), "0x");
+    }
+
+    function performUpkeep(bytes calldata /* performData */) external {
+        Require.that(
+            msg.sender == CHAINLINK_REGISTRY,
+            _FILE,
+            "caller is not Chainlink"
+        );
+        Require.that(
+            _checkUpkeepConditions(),
+            _FILE,
+            "checkUpkeep conditions not met"
+        );
+
+        _updateExchangeRateAndTimestamp();
+    }
+
+    // ============================ Internal Functions ============================
+
+    function _ownerSetHeartbeat(uint256 _heartbeat) internal {
+        emit HeartbeatSet(_heartbeat);
+        HEARTBEAT = _heartbeat;
+    }
+
+    function _ownerSetGracePeriod(uint256 _gracePeriod) internal {
+        emit GracePeriodSet(_gracePeriod);
+        GRACE_PERIOD = _gracePeriod;
+    }
+
+    function _ownerSetUpperEdge(uint256 _upperEdge) internal {
+        Require.that(
+            _upperEdge > 10000,
+            _FILE,
+            "Invalid upper edge"
+        );
+        emit UpperEdgeSet(_upperEdge);
+        UPPER_EDGE = _upperEdge;
+    }
+
+    function _ownerSetLowerEdge(uint256 _lowerEdge) internal {
+        Require.that(
+            _lowerEdge < 10000,
+            _FILE,
+            "Invalid lower edge"
+        );
+        emit LowerEdgeSet(_lowerEdge);
+        LOWER_EDGE = _lowerEdge;
+    }
+
+    function _ownerSetChainlinkRegistry(address _chainlinkRegistry) internal {
+        Require.that(
+            _chainlinkRegistry != address(0),
+            _FILE,
+            "Invalid chainlink registry"
+        );
+        emit ChainlinkRegistrySet(_chainlinkRegistry);
```

Comment:
Is there a view function you can call on the registry to ensure this address was set properly?